### PR TITLE
fix: Always prefer tauri configs further up the file tree

### PR DESCRIPTION
**Pull request by FabianLars**
This fixes issues where the action used config files in deeply nested directories instead of the correct one in the upper levels.

ref: https://discord.com/channels/616186924390023171/1119317689836515388

kinda quick and dirty, have to call it a day. If it works for the user in that thread too i'll try to clean it up a bit. (or maybe not as i don't want to have a helper function just to reuse that inner for loop)

Edit: User confirmed it worked for them, therefore converting it to a normal PR, still gonna plan to try to clean it up a bit but feel free to have a look.

### CODE REVIEW

**Comment by FabianLars** [comment]
okay nevermind about cleaning it up, i'd rather keep it as simple&stupid as possible for now.

**Review by amrbashir — COMMENTED** [review]
Looking at the code, it seems that this is only traversing down and not up, right? if so I fail to see how is this different from the previous impl

**Comment by FabianLars** [comment]
take for example this file tree, taken from https://github.com/tonymushah/special-eureka/tree/master/src-tauri:
```
/
  src-tauri/
    examples/
      tauri.conf.json
    src/
    tauri.conf.json # this is the file we want
```

The current implementation will always select src-tauri/examples/tauri.conf.json while the new one will go for src-tauri/tauri.conf.json -> tested by me and confirmed by 2 community members one of which linked above.

One variant i didn't think about would be this
```
/
  src/
  some-package/
    whatever/
      tauri.conf.json
  src-tauri/
    tauri.conf.json # this is the file we want
```

Here the new implementation will also go for the wrong file. I can't remember if setting projectPath to the src-tauri dir and not its parent works too so that would need some testing too ig

**Comment by amrbashir** [comment]
I see but as you outlined this fix is not complete as it doesn't account for levels at all, I'd recommend reverting this change and use `glob` again but instead of taking the first result, we sort the return paths first by how far are they to the current working directory.

**Comment by FabianLars** [comment]
Do you have any pointers on how to sort them then? Because i didn't find an easy-ish & reliable way to do that which is really why this PR exists in the first place.

**Comment by amrbashir** [comment]
if we are looking down only, then it should be easy to just split on path separator and count the components of the path, and choose the lowest number

**Comment by FabianLars** [comment]
yeah, that sounded kinda scary in my head but if you also have that in mind we can try it out and keep an eye on it, thanks 👍

**Comment by FabianLars** [comment]
I found an old todo entry to evaluate globby since it can handle gitignore files dynamically (i mean it adds gitignore files along the way, not just the ones we give it upfront) so i switched to that. And apparently it already sorts the results correctly in contrast to really all the other glob/file search packages i tried 🥳 

I'll keep an eye on it though and re-add the `sort()` call if necessary.

File: .changes/enhanve-tauridir-search-gitignore.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+action: patch:enhance
```
Comment (amrbashir) [inline .changes/enhanve-tauridir-search-gitignore.md]: note that this repo does not have tags configured, so the final release will have a `# enhance` header where I think you wanted to have `# Enhancements`. I think we don't use the covector tag features for small repos like this one.

File: src/utils.ts [hunk 2]
```diff
@@ -54,29 +53,27 @@ export function getPackageJson(root: string) {
...
-  const ignoreRules = ignore();
-  const gitignorePath = join(root, '.gitignore');
-  if (existsSync(gitignorePath)) {
-    ignoreRules.add(readFileSync(gitignorePath).toString());
-  } else {
-    ignoreRules.add('**/node_modules').add('**/target');
-  }
-  const paths = globSync(
+  const tauriConfPaths = globbySync(
     ['**/tauri.conf.json', '**/tauri.conf.json5', '**/Tauri.toml'],
```
Comment (amrbashir) [inline src/utils.ts:63]: do we want to also consider platform-specific config files?

**Review by amrbashir — APPROVED** [review]

File: .changes/enhanve-tauridir-search-gitignore.md (hunk 1 above)
Comment (FabianLars) [reply .changes/enhanve-tauridir-search-gitignore.md]: Okay, i'll just remove the tags then. thanks for the headsup

File: src/utils.ts (hunk 2 above)
Comment (FabianLars) [reply src/utils.ts:63]: iirc tauri itself requires one of these files to be present and can't work with just the platform specific files so i would not add them here either.

Comment (amrbashir) [reply src/utils.ts:63]: cool

**Review by amrbashir — APPROVED** [review]

**Pull request merged**